nip47/nip47_service.go

```diff
@@ -29,8 +28,10 @@ type Nip47Service interface {
 	events.EventSubscriber
 	StartNotifier(ctx context.Context, relay *nostr.Relay, lnClient lnclient.LNClient)
 	HandleEvent(ctx context.Context, relay nostrmodels.Relay, event *nostr.Event, lnClient lnclient.LNClient)
-	PublishNip47Info(ctx context.Context, relay nostrmodels.Relay, lnClient lnclient.LNClient) error
-	CreateResponse(initialEvent *nostr.Event, content interface{}, tags nostr.Tags, ss []byte) (result *nostr.Event, err error)
+	GetNip47Info(ctx context.Context, relay *nostr.Relay, appWalletPubKey string) (*nostr.Event, error)
+	PublishNip47Info(ctx context.Context, relay nostrmodels.Relay, appWalletPubKey string, appWalletPrivKey string, lnClient lnclient.LNClient) (*nostr.Event, error)
+	PublishNip47InfoDeletion(ctx context.Context, relay nostrmodels.Relay, appWalletPubKey string, appWalletPrivKey string, infoEventId string) error
```

Comment:
Maybe we could, but since we already need/use `appWalletPubKey` in other places "outside" (before calling the service) and we have it already calculated , it would cause us to unnecessarily do multiple times `nostr.GetPublicKey(walletPrivKey)` which is not ideal IMO. Wdyt?

Reply:
True, just wanted to align things. But if we introduce pubkey for `CreateResponse` it would be additional code for legacy case. Best to leave it this way 👍